bus-mapping/src/circuit_input_builder.rs

```diff
@@ -875,31 +1136,83 @@ impl<'a> CircuitInputBuilder {
         eth_tx: &eth_types::Transaction,
         geth_trace: &GethExecTrace,
     ) -> Result<(), Error> {
-        let mut tx = self.new_tx(eth_tx)?;
+        // Iterate over geth_trace to inspect and collect each call's
+        // is_success, which is at the top of stack at the step after a
+        // call.
+        let mut call_is_success = HashMap::new();
+        let mut call_indices = Vec::new();
+        for (index, geth_step) in geth_trace.struct_logs.iter().enumerate() {
+            if let Some(geth_next_step) = geth_trace.struct_logs.get(index + 1)
+            {
+                // Dive into call
+                if geth_step.depth + 1 == geth_next_step.depth {
+                    call_indices.push(index);
+                // Emerge from call
+                } else if geth_step.depth - 1 == geth_next_step.depth {
+                    let is_success = !geth_next_step.stack.last()?.is_zero();
+                    call_is_success
+                        .insert(call_indices.pop().unwrap(), is_success);
+                // When callee doesn't have code for execution, the depth won't
+                // chagne, so we need to catch such cases here.
+                } else if matches!(
```

Comment:
Ah, I got confused by myself. No context should be generated if callee doesn't have code to be executed.

Tho I used to think we could treat "callee without code" as "callee with code having a single `STOP`", since the behavior is same due to the implicit `STOP` (yellowpaper also doesn't explicitly say how a empty code will be executed, or skipped). Then we can save some constraints in circuit with only 1 extra `STOP` step.

But it seems more straightforward to just not dive into the call and continue to current context's next opcode.